### PR TITLE
Transformers OMLflowCallback

### PR DESCRIPTION
**Pull request by twolffpiggott**
Relates to issue #52 

Defines an `integration` section within `mltb` with a Transformers `OMLflowCallback` object.

Implemented to not introduce a strict dependency on the Transformers package using the [deferred import exception convention defined by Optuna](https://github.com/optuna/optuna/blob/78172ee94b447eaf70feb0f47c54dc334b3c63b8/optuna/_imports.py).

### CODE REVIEW

**Comment by PhilipMay** [comment]
Hey @twolffpiggott thanks for this PR. :+1: 

I do not understand the Mock implementation. IMO if someone uses this script (imports the module) without having transformers installed - it should just fail and quit working. What do you think?

**Comment by PhilipMay** [comment]
Maybe you also could add a test? This might help as a template: https://github.com/PhilipMay/mltb/blob/master/tests/test_omlflow.py

File: mltb/integration/transformers.py [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+
+        Args:
+            trial: OptunaMLflow object
+            log_training_args: Whether to log all Transformers TrainingArguments as MLflow params
+            log_model_config: Whether to log the Transformers model config as MLflow params
+
+        Raises:
+            ImportError: If the Transformers package is not installed correctly
+        """
+        _imports.check()
```
Comment (PhilipMay) [inline mltb/integration/transformers.py]: Is this redundant to the same call in line 11?

**Comment by PhilipMay** [comment]
Hey @twolffpiggott I am sorry for the delay.
Many thanks again.

I approved that CI is allowed to run. There are some errors.
Can you please check them?
I also added an inline comment.

**Review by PhilipMay — COMMENTED** [review]
Thanks. Can you please check the inline comment?

File: mltb/integration/transformers.py (hunk 1 above)
Comment (twolffpiggott) [reply mltb/integration/transformers.py]: Nice spot, I agree. I've removed the in-class check.

**Comment by PhilipMay** [comment]
Many thanks for thos PR @twolffpiggott 👍

**Pull request merged**